Feature/log in

michellengnx opened the pull request:
**_What this PR achieves:_**

🌟 **Documentation**
- Create folder `documentation/design-decisions` with template to document your decision as you implement your use case. Idea stolen from Spotify's very own Architecture Decision Record. Read more about it [here](https://engineering.atspotify.com/2020/04/when-should-i-write-an-architecture-decision-record/).
- Add `TODOs` on several entity classes about attribute types and deprecated import which will need to be revisited and possibly changed going forward. 

🌟 **Testing** 
- Create more unit tests for `CreditCard`, `EmergencyContact`, and `Patient` class, which will be used in implementing `login` use case.

🌟 **Cleanup** 
- Who doesn't love a tidy repo?

armaan-n reviewed (changes requested):
Changes to the entities and the addition of documentation looks fine. However, you should probably refrain from using `assert` in you test classes. To remain consistent with JUnit's API (and the rest of our code base), stick to using the assertion statements provided by Junit (e.g. `assertEquals(a, b)` instead of `assert a.equals(b)`). I'm also not sure whether you added tests to `PatientTest.java` (unless you're just making sure the constructor runs without crashing). These modifications should be made before merging with `develop`.

michellengnx commented:
Great catch on the PatientTest class. I in fact did forget the assert statements. Added now!
Other changes made:
- Seems like we are using different version of junit. I removed junit 4 dependency in pom.xml and we should be using junit 5 moving forward for consistency's sake. There are minor syntax different between the two version. Also cc: @ISMAELCHONA04 @JustinHarvo @izorakayumova so that you guys are aware.
-  Temporarily removed docker job on workflow.

armaan-n reviewed (approved):
Issues from last pull request have been resolved.
